box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaboratorsRolesFragment.java

```diff
@@ -1,64 +1,103 @@
+import com.box.androidsdk.share.vm.PresenterData;
 import com.box.androidsdk.share.vm.SelectRoleShareVM;
+import com.box.androidsdk.share.vm.ShareVMFactory;
+
 
+public class CollaboratorsRolesFragment extends Fragment {
 
-public class CollaboratorsRolesFragment extends Fragment implements FragmentTitle {
 
+    private void setTitles() {
+        ActionbarTitleVM actionbarTitleVM = ViewModelProviders.of(getActivity()).get(ActionbarTitleVM.class);
+        actionbarTitleVM.setTitle(getString(R.string.box_sharesdk_title_access_level));
+        actionbarTitleVM.setSubtitle(null);
+    }
 
     public interface RoleUpdateNotifier {
         void setRole(BoxCollaboration.Role role);
+
+        void notifyRemove();
     }
 
     public static final String TAG = CollaboratorsRolesFragment.class.getName();
     SelectRoleShareVM vm;
 
+
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         UsxFragmentCollaborationRolesBinding binding = DataBindingUtil.inflate(inflater, R.layout.usx_fragment_collaboration_roles, container, false);
         View view = binding.getRoot();
 
+        setTitles();
+
         vm = ViewModelProviders.of(getActivity()).get(SelectRoleShareVM.class);
         binding.setViewModel(vm);
-        binding.setRoleUpdateNotifier(vm::setSelectedRole);
-        return view;
-    }
+        binding.setRoleUpdateNotifier(new RoleUpdateNotifier() {
+            @Override
+            public void setRole(BoxCollaboration.Role role) {
+                vm.setSelectedRole(role);
+            }
+
+            @Override
+            public void notifyRemove() {
+                showRemoveWarning();
+            }
+        });
 
-    @Override
-    public int getFragmentTitle() {
-        return R.string.box_sharesdk_title_access_level;
+        return view;
     }
 
-
-    @Override
-    public void onSaveInstanceState(@NonNull Bundle outState) {
-        super.onSaveInstanceState(outState);
+    public static CollaboratorsRolesFragment newInstance(){
+        return new CollaboratorsRolesFragment();
     }
-
-
-    @Override
-    public int getFragmentSubtitle() {
-        return -1;
+    private void showRemoveWarning() {
+        String deleteDifferentWarning = getResources().getString(R.string.box_sharesdk_warn_remove_different_collaboration_folder, vm.getName(),vm.getCollaboration().getItem().getName());
+        AlertDialog dialog = new AlertDialog.Builder(getActivity()).setTitle(R.string.box_sharesdk_title_remove_different_collaboration_folder)
+                .setMessage(deleteDifferentWarning)
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(final DialogInterface dialog, final int which) {
+                        vm.setRemoveSelected(true);
+                        getActivity().onBackPressed();
```

Comment:
If you are just trying to dismiss the alert then you should call `dialog.dismiss()` or `dialog.cancel()`

Reply:
I am not only dismissing the dialog. I am also switching back to the previous fragment.